src/main/java/net/neoforged/neoforge/client/event/RegisterDimensionTransitionScreenEvent.java

```diff
@@ -0,0 +1,68 @@
+import net.neoforged.neoforge.client.DimensionTransitionScreenManager.ReceivingLevelScreenFactory;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * <p>Event for registering screen effects when transitioning across dimensions.
+ * Note that there is a priority order when it comes to what screens are displayed: <br>
+ * - Using registerConditionalEffect has priority over the usual transition effects, and will only fire when travelling to the specified dimension coming from a certain dimension. <br>
+ * - If a dimension has a screen that displays when entering it, that will have priority over a dimension that has one when you leave it. </p>
+ *
+ * <p>This event is not {@linkplain ICancellableEvent cancellable}, and does not {@linkplain HasResult have a result}.
+ *
+ * <p>This event is fired on the mod-specific event bus, only on the {@linkplain LogicalSide#CLIENT logical client}.</p>
+ */
+public class RegisterDimensionTransitionScreenEvent extends Event implements IModBusEvent {
+    private final Map<Pair<ResourceKey<Level>, ResourceKey<Level>>, ReceivingLevelScreenFactory> conditionalDimensionEffects;
+    private final Map<ResourceKey<Level>, ReceivingLevelScreenFactory> toEffects;
+    private final Map<ResourceKey<Level>, ReceivingLevelScreenFactory> fromEffects;
+
+    @ApiStatus.Internal
+    public RegisterDimensionTransitionScreenEvent(Map<Pair<ResourceKey<Level>, ResourceKey<Level>>, ReceivingLevelScreenFactory> conditionalDimensionEffects,
+            Map<ResourceKey<Level>, ReceivingLevelScreenFactory> toEffects,
+            Map<ResourceKey<Level>, ReceivingLevelScreenFactory> fromEffects) {
+        this.conditionalDimensionEffects = conditionalDimensionEffects;
+        this.toEffects = toEffects;
+        this.fromEffects = fromEffects;
+    }
+
+    /**
+     * Registers a dimension transition when traveling to a dimension.
+     */
+    @Nullable
+    public ReceivingLevelScreenFactory registerIncomingEffect(ResourceKey<Level> dimension, ReceivingLevelScreenFactory screen) {
+        return this.toEffects.putIfAbsent(dimension, screen);
+    }
+
+    /**
+     * Registers a dimension transition when traveling from a dimension.
+     */
+    @Nullable
+    public ReceivingLevelScreenFactory registerOutgoingEffect(ResourceKey<Level> dimension, ReceivingLevelScreenFactory screen) {
+        return this.fromEffects.putIfAbsent(dimension, screen);
+    }
+
+    /**
+     * Registers a dimension transition when traveling to a dimension from a certain dimension.
+     * This registration method takes priority over the normal to and from dimension checks.
+     */
+    @Nullable
+    public ReceivingLevelScreenFactory registerConditionalEffect(ResourceKey<Level> toDimension, ResourceKey<Level> fromDimension, ReceivingLevelScreenFactory screen) {
+        return this.conditionalDimensionEffects.putIfAbsent(Pair.of(toDimension, fromDimension), screen);
+    }
```

Comment:
```suggestion
    /**
     * Registers a dimension transition when traveling to a dimension from a certain dimension.
     * This registration method takes priority over the normal to and from dimension checks.
     *
     * @returns <code>true</code> if the screen was registered, <code>false</code> otherwise.
     */
    public boolean registerConditionalEffect(ResourceKey<Level> toDimension, ResourceKey<Level> fromDimension, ReceivingLevelScreenFactory screen) {
        return this.conditionalDimensionEffects.putIfAbsent(Pair.of(toDimension, fromDimension), screen) == null;
    }
```